internal/mailbox/balance.go

```diff
@@ -0,0 +1,25 @@
+// Copyright 2019 Finobo
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mailbox
+
+import (
+	"context"
+)
+
+//go:generate mockgen -source=balance.go -package=mailboxtest -destination=./mailboxtest/balance_mock.go
+// Receiver gets balance from blockchain.
+type BalanceFinder interface {
+	GetBalance(ctx context.Context, protocol, network string, address []byte) (string, error)
```

Comment:
`string` as a return type will mean it needs to be converted before it can be used. How about using something like `uint64` to represent the smallest unit, e.g. wei